sign.go

```diff
@@ -443,12 +468,41 @@ func (m *SignMessage) Sign(rand io.Reader, external []byte, signers ...Signer) e
 // Notice: The COSE Sign API is EXPERIMENTAL and may be changed or removed in a
 // later release.
 func (m *SignMessage) Verify(external []byte, verifiers ...Verifier) error {
+	return m.verify(nil, external, verifiers...)
+}
+
+// VerifyDetached verifies the signatures on the SignMessage against the corresponding
+// verifier, returning nil on success or a suitable error if verification fails.
+//
+// Returns ErrMissingPayload if detached is nil, and ErrMultiplePayloads if the SignMessage
+// also has an attached payload.
+//
+// See `Signature.Verify()` for advanced verification scenarios like threshold
+// policies.
+//
+// Reference: https://datatracker.ietf.org/doc/html/rfc8152#section-4.4
+//
+// # Experimental
+//
+// Notice: The COSE Sign API is EXPERIMENTAL and may be changed or removed in a
+// later release.
```

Comment:
Is Line 487 and 488 relevant here ?, given that we are commenting on Verify here?

Reply:
I copied these lines from the existing methods. I took it to mean the whole of the Sign/Verify API together.

Reply:
Perhaps we should change this to sign/verify

Reply:
I'm waiting to see if anyone else has strong opinions on this.

Reply:
This is a nit, I suggest filing an issue and addressing is a subsequent PR and resolving this.

Reply:
Resolving this comment copy/paste nit. Please open a new issue/PR if needed.